### ISSUE TITLE
rodio::Source not satisfied error on implementation identical to example

### ISSUE THREAD

**Issue by micro-80** [probablykasper/redlux#3]
Hello,

I'm writing a program with the hope that I can use redlux with rodio, and so I looked at the examples and just copied and pasted the m4a example to see how it worked - however, I can't seem to get it to compile, with the error 

> let decoder = Decoder::new_mpeg4(buf, size).expect("Error creating M4aDecoder");
>the trait bound `redlux::Decoder<std::io::BufReader<std::fs::File>>: rodio::Source` is not satisfied
>
>the trait `rodio::Source` is not implemented for `redlux::Decoder<std::io::BufReader<std::fs::File>>`rustc(E0277)
>mod.rs(19, 17): the trait `rodio::Source` is not implemented for `redlux::Decoder<std::io::BufReader<std::fs::File>>`

Always appearing on the `sink.append(decoder)` line. More specifically, this is my implementation (essentially a copy/paste, but with an argument instead of a hard-coded path)
```rust
use redlux::Decoder;
use rodio::{OutputStream, Sink};
use std::fs::File;
use std::io::BufReader;

fn play_mp4(path: &String) {
    let file = File::open(path).expect("Error opening file");

    let metadata = file.metadata().expect("Error getting file metadata");
    let size = metadata.len();
    let buf = BufReader::new(file);

    let decoder = Decoder::new_mpeg4(buf, size).expect("Error creating M4aDecoder");

    let output_stream = OutputStream::try_default();
    let (_stream, handle) = output_stream.expect("Error creating output stream");
    let sink = Sink::try_new(&handle).expect("Error creating sink");

    sink.append(decoder);
    sink.play();
    sink.set_volume(0.25);
    sink.sleep_until_end();
} 
```

**Comment by kasper9n** [comment]
The reason is that redlux depends on rodio 0.13.0, so using it with a version like 0.14.0 gives this error.

Not 100% sure if I should just publish a patch that adds rodio `0.14.*` or something else

**Comment by kasper9n** [comment]
Fixed in 43b83580fdc68ad79e145ec26d5c1fc57148d0a5

**Comment by micro-80** [comment]
Appreciate the quick response, thank you :)
